iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -453,7 +456,13 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
         commitWrite(table, branchName, snapshotId, startTime, filesForCommit, operation);
       }
     } else {
-      commitOverwrite(table, branchName, startTime, filesForCommit);
+
+      RewritePolicy rewritePolicy = RewritePolicy.fromString(outputTable.jobContexts.stream()
+          .findAny()
+          .map(x -> x.getJobConf().get(ConfVars.REWRITE_POLICY.varname, RewritePolicy.DEFAULT.name()))
```

Comment:
why do you define defaults twice? you already have `orElse`

Reply:
Fixed